### PR TITLE
constructable rechargers

### PR DESCRIPTION
**Pull request by Segonist**
## About the PR

Now you can make cell recharger and recharger like other machines.
- Added circuits for cell recharger and recharger, they can be made in autholathe
- You can construct them like other machines, with 2 capacitors, and 5 lv cables for cell recharger and 5 mv cables for recherger.
- They can be upgraded (I don't think I made good formula for charging speed, correct me if so)
- Wall recharger wouldn't have it's construction graph, because I can't made it :(

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/74138556/fee3467e-b30e-4172-9194-fb8d18b9027b

**Changelog**

:cl: Segonist
- add: Now you can print cell recharger and recharger circuits on autolathe and construct them like other machines.

### CODE REVIEW

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 1]
```diff
@@ -428,6 +428,8 @@
   - IndustrialEngineering
   unlockedRecipes:
   - PowerDrill
+  - CellRechargerCircuitboard
+  - WeaponCapacitorRechargerCircuitboard
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:432]: these don't need to be a technology. just put them on the autolathe.

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 2]
```diff
@@ -447,6 +447,38 @@
...
+  name: cell recharger machine board
+  description: A machine printed circuit board for a cell recharger.
+  components:
+    - type: Sprite
+      sprite: Objects/Misc/module.rsi
+      state: charger_APC
+    - type: MachineBoard
+      prototype: PowerCellRecharger
+      requirements:
+        Capacitor: 2
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:462]: don't use machine parts unless you're implementing upgrades. Mind you, upgrades would be less than 100 lines of code, so do consider it.

**Comment by Segonist** [comment]
How can I edit this About the PR message?

**Comment by deltanedas** [comment]
it feels wierd to be in autolathe when every other board in the game is in protolathe or circuit imprinter, even without technology requirement.

**Comment by Segonist** [comment]
> it feels wierd to be in autolathe when every other board in the game is in protolathe or circuit imprinter, even without technology requirement.

should I just put them in the circuit imprinter?

**Comment by deltanedas** [comment]
i think some like apc electronics are in both so maybe put in protolathe and circuit imprinter

**Comment by Segonist** [comment]
> i think some like apc electronics are in both so maybe put in protolathe and circuit imprinter

wait, after rnd rework all boards from the protolathe were moved to the autolathe

**Comment by Segonist** [comment]
```
Error Message:
   Multiple failures or warnings in test:
  1) Prototype PowerCellRecharger modifies component on spawn: ContainerContainer. Modified fields: containers
  2) Prototype WeaponCapacitorRecharger modifies component on spawn: ContainerContainer. Modified fields: containers
```
what does it mean and how to fix it?

**Comment by deltanedas** [comment]
if you are using basemachine its probably double creating containers idk

File: Content.Server/Power/Components/ChargerComponent.cs [hunk 3]
```diff
@@ -8,8 +10,32 @@ public sealed class ChargerComponent : Component
         [ViewVariables]
         public CellChargerStatus Status;
 
+        /// <summary>
+        /// The charge rate of the charger
```
Comment (deltanedas) [inline Content.Server/Power/Components/ChargerComponent.cs:14]: pro useless comment
at least say `, in watts`

File: Resources/Locale/en-US/power/components/charger.ftl [hunk 4]
```diff
@@ -1 +1,2 @@
-charger-examine = Charges [color={$color}]{$chargeRate}W[/color] per second.
\ No newline at end of file
+charger-examine = Charges [color={$color}]{$chargeRate}W[/color] per second.
```
Comment (deltanedas) [inline Resources/Locale/en-US/power/components/charger.ftl:1]: change W to J or remove per second we are big boys and know what watts are :trollface:

File: Resources/Prototypes/Entities/Structures/Power/chargers.yml [hunk 5]
```diff
@@ -93,10 +98,76 @@
 - type: entity
   name: wall recharger
   id: WallWeaponCapacitorRecharger
-  parent: WeaponCapacitorRecharger
+  placement:
+    mode: SnapgridCenter
   components:
   - type: Sprite
+    netsync: false
```
Comment (DawBla) [inline Resources/Prototypes/Entities/Structures/Power/chargers.yml:105]: Sprite component is client only so no need for netsync: false
https://discord.com/channels/310555209753690112/928056457117696020/1102795784673579149

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by EmoGarbage404** [comment]
These boards are good to be the autolathe under the doctrine of:  tech fpr basic shit you find on-station is dumb

File: Resources/Locale/en-US/power/components/charger.ftl (hunk 4 above)
Comment (Segonist) [reply Resources/Locale/en-US/power/components/charger.ftl:1]: what... why? It charges watts per second, isn't it?

**Comment by deltanedas** [comment]
you find apcs, noors, firelocks, etc too
maybe all them should be in autolathe too

**Comment by deltanedas** [comment]
im 99% sure they arent i checked like 2 hours ago

**Comment by Segonist** [comment]
@metalgearsloth this pr is ready for review

File: Resources/Locale/en-US/power/components/charger.ftl (hunk 4 above)
Comment (deltanedas) [reply Resources/Locale/en-US/power/components/charger.ftl:1]: watts is joules per second bruh

**Pull request merged**